decidim-core/app/helpers/decidim/filters_helper.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+module Decidim
+  # Helper that provides a single method to create filter resource forms
+  module FiltersHelper
+    # This method wraps everything in a div with class filters and calls
+    # the form_for helper with a custom builder
+    #
+    # filter - A filter object
+    # block  - A block to be called with the form builder
+    #
+    # Returns the filter resource form wrapped in a div
+    def filter_form_for(filter)
+      filters_container = content_tag :div, class: "filters" do
+        form_for filter, builder: FilterFormBuilder, url: url_for, as: :filter, method: :get, remote: true do |form|
+          yield form
+        end
+      end
+      filters_container + javascript_include_tag("decidim/filters")
```

Comment:
I didn't find any issue so far. We are dealing with turbolinks events to bind and unbind change form events. We also use history API to handle the back and forward buttons and update the form values as well.